add a consequence of AC

DanGrayson opened the pull request:
Theorem SetCovering (X:Type) : AxiomOfChoice -> ∃ (S:hSet) (f:S->X), issurjective f.

tpannila commented:
I would write this PR as follows
```
(** Idea of the proof.
   - The proof uses the weak equivalence [weqpathsinsetquot] together with the equivalence relation
     [pathseqrel X]. These give us the following weak equivalence
                              [x = x' ≃ (setquotpr R x = setquotpr R x')]
   - By [AxiomOfChoice_surj] there exists a surjective map [g : setquotinset (pathseqrel X) → X]
     such that
               [∀ x : setquotinset (pathseqrel X), setquotpr (pathseqrel X) (g x) = x].
     Hence
         [setquotpr (pathseqrel X) (g (setquotpr (pathseqrel X) x) = setquotpr (pathseqrel X) x]
     and by the above weak equivalence
                            [g (setquotpr (pathseqrel X) x) = x].
     This shows that g is surjective, because now [setquotpr (pathseqrel X) x] gives an 
     [hfiber g x].
*)
Theorem SetCovering (X : UU) : AxiomOfChoice -> ∃ (S : hSet) (f : S -> X), issurjective f.
Proof.
  intros AC.
  use (squash_to_prop
         ((dirprod_pr1 AC_impl2 AC) (setquotinset (pathseqrel X)) X (setquotpr (pathseqrel X))
                                    (issurjsetquotpr (pathseqrel X)))
         (propproperty _)).
  intros f. use hinhpr. use tpair.
  - exact (setquotinset (pathseqrel X)).
  - use tpair.
    + exact (pr1 f).
    + intros x.
      use (squash_to_prop
             (invmap (weqpathsinsetquot (pathseqrel X) (pr1 f ((setquotpr (pathseqrel X)) x)) x)
                     (pr2 f (setquotpr (pathseqrel X) x)))
             (propproperty _)).
      intros e. use hinhpr. use hfiberpair.
      * exact (setquotpr (pathseqrel X) x).
      * exact e.
Qed.
```

DanGrayson commented:
You're right, that lemma missed the boat.  I've simplified the proof, provided a more suitable
definition of pi0, and added a comment, but preferred a comment addressing the mathematics
rather than the code, since the code speaks for itself.

DanGrayson commented:
@benediktahrens ?

benediktahrens commented:
lgtm.
the comment would be more useful if it were also shown in the html - in particular since you are already using the proper markup for the identifier.

DanGrayson commented:
oops, thanks, I've fixed the comment.

DanGrayson commented:
I cancelled the travis build, since only a comment was changed.
